### PR TITLE
Unify @BindBean and SqlStatement.bindFromProperties

### PR DESCRIPTION
**Pull request by qualidafial**
* Enhance BeanPropertyArguments to support bean name prefix like `@BindBean`
* Rename `SqlStatement.bindFromProperties()` to `.bindBean()`
* Rename `SqlStatement.bindFromMap()` to `.bindMap()`
* Modify `@BindBean` to use `SqlStatement.bindBean` instead of its own parallel implementation.
  This also makes `@BindBean` bindings lazy instead of eager, which fixes issue #478
* Modify @BindMap:
  * Rename `BindMap.value()` to `.keys()`
  * Rename `BindMap.prefix()` to `.value()` (to agree with `@BindBean`)
  * Rename `BindMap.implicitKeyStringConversion()` to `.convertKeys()`
  * Use `SqlStatement.bindMap` instead of its own parallel implementation.

### CODE REVIEW

**Comment by qualidafial** [comment]
Fixes #494, #478

**Comment by qualidafial** [comment]
@arteam Would you mind reviewing this PR?

File: core/src/main/java/org/jdbi/v3/core/BeanPropertyArguments.java [hunk 1]
```diff
@@ -26,53 +26,54 @@
...
-                                                                             name, bean), e, ctx);
+                    try
+                    {
+                        return ctx.findArgumentFor(
+                                descriptor.getReadMethod().getGenericReturnType(),
+                                descriptor.getReadMethod().invoke(bean));
+                    }
+                    catch (IllegalAccessException e)
+                    {
+                        throw new UnableToCreateStatementException(String.format("Access exception invoking getter for " +
```
Comment (arteam) [inline core/src/main/java/org/jdbi/v3/core/BeanPropertyArguments.java]: Could we write a test for this exception?

File: core/src/main/java/org/jdbi/v3/core/SqlStatement.java [hunk 2]
```diff
@@ -308,31 +308,41 @@ public SelfType bind(String name, Argument argument)
...
      */
-    public SelfType bindFromProperties(Object o)
+    public SelfType bindBean(Object bean)
     {
-        return bindNamedArgumentFinder(new BeanPropertyArguments(o, getContext(), getArgumentRegistry()));
+        return bindNamedArgumentFinder(new BeanPropertyArguments(null, bean, getContext()));
+    }
+
+    /**
+     * Binds named parameters from JavaBean properties on o, with the given prefix.
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/SqlStatement.java]: Done

File: core/src/main/java/org/jdbi/v3/core/BeanPropertyArguments.java (hunk 1 above)
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/BeanPropertyArguments.java]: Done

**Pull request merged**